hishel/_async/_transports.py

```diff
@@ -181,13 +181,24 @@ async def handle_async_request(self, request: Request) -> Response:
 
                 # Merge headers with the stale response.
                 final_httpcore_response = self._controller.handle_validation_response(
-                    old_response=stored_response, new_response=httpcore_revalidation_response
+                    old_response=stored_response,
+                    new_response=httpcore_revalidation_response,
                 )
 
                 await final_httpcore_response.aread()
                 await revalidation_response.aclose()
 
                 assert isinstance(final_httpcore_response.stream, tp.AsyncIterable)
+
+                if self._controller.is_cachable(
+                    request=httpcore_request,
+                    response=final_httpcore_response,
+                ):
+                    await self._storage.store(
```

Comment:
I guess we need to check, if the response code is 200, then call `store` with the new metadata (by passing metadata=None so it creates automatically) but if the response code is 304 then we should use `update_metadata` method, which is for updating the entries, without touching the `ttl`. No we are reseting the metadata with each 304, which is not what we want

Reply:
Ok, I see that _create_hishel_response already does the "update_metadata" call so I simply exclude 304 from the condition to store revalidation response, as it is indeed already done in this case. But maybe if the _create_hishel_response already have the responsibility of updating stored entry, the storage.store calls would benefit from being in the same place ?